internal/test/daemon/daemon.go

```diff
@@ -482,6 +490,7 @@ func (d *Daemon) Restart(t testingT, args ...string) {
 // RestartWithError will restart the daemon by first stopping it and then starting it.
 func (d *Daemon) RestartWithError(arg ...string) error {
 	if err := d.StopWithError(); err != nil {
+		d.log.Logf("[%s] Error when stopping daemon: %v", d.id, err)
```

Comment:
`d.StopWithError()` has a call to `d.logFile.Close()` in defer so in most cases this statement won't log anything. I suggest moving it to the above mentioned defer, e.g. something like:
```diff
-func (d *Daemon) StopWithError() error {
+func (d *Daemon) StopWithError() err error {
         if d.cmd == nil || d.Wait == nil {
                 return errDaemonNotStarted
         }
         defer func() {
-                d.log.Logf("[%s] Daemon stopped", d.id)
+               if err == nil {
+                        d.log.Logf("[%s] Daemon stopped", d.id)
+               } else {
+                        d.log.Logf("[%s] Error stopping daemon: %v", d.id, err)
+               }
                 d.logFile.Close()
                 d.cmd = nil
         }()
```

Reply:
Nice catch, fixed!